### DISCUSSION TITLE
Need some talk around #4424 issue.

### DISCUSSION THREAD

**Discussion by HyperDanisH** [monkeytypegame/monkeytype#4466] [Ideas] [answered]
Hey everyone,

I have been assigned to resolve issue #4424, I need some ideas about how how should I approach this thing so that I can speed up the process.

First of all, I have a UI change to impliment this feature, it is from:

![Screenshot 2023-07-19 164924](https://github.com/monkeytypegame/monkeytype/assets/93310647/4b209571-4bf7-4810-94d4-7ca4da84f393)

this to this:

![Screenshot 2023-07-19 164003](https://github.com/monkeytypegame/monkeytype/assets/93310647/dcc4d3f3-2170-438f-aa6b-0f1a933ea088)

and I need some suggestions on how I can improve this.

Secondly, I have to get the min speed required from backend, and I want to know how I can achieve that. For example, tell me what should be the min speed and is it already stored somewhere in database or would I have to impliment that myself.

Best,

**Comment by Miodec** [comment, marked as answer]
For the UI i would make it part of the same string, so something like `Not qualified (min speed required: 100wpm)`

For the backend, you can take a look at how the `getRank()` function works, and modify to get the min speed. You will need to google around to find code that gets the minimum value from a Redis collection.

**Reply by HyperDanisH** [reply]
Hii,

I have been searching for a way to sort redis hash based on wpm of user, but I don't think there is a way out of the box using redis cli or api. So, should I use a custom sorting algoridhm written in lua script? If so then what kind of sorting algo should I be using here (any prefrence).

Secondly, I wanted to ask, How should I determine min rank? Should it be based upon wpm or raw wpm.

Best,

**Reply by Miodec** [reply]
Are you sure there is no way to just get the minimum value?

**Reply by Miodec** [reply]
Just take the min value from the scores 
![image](https://github.com/monkeytypegame/monkeytype/assets/13181393/277a3d85-5908-4a18-84e0-c050158bb5d3)

They are encoded as follows:

```
1ZZZXXCCCVVBBBBB

Z - wpm
X - two decimal places for wpm
C - accuracy
V - two decimal places for accuracy
B - timestamp information
```

Just take the lowest score, extract the wpm out of it (maybe round or ceil it) and return.

**Reply by Miodec** [reply]
Or, you could also get the lowest score, take the value column (user uid) then get the result object from the results list (by parsing the json)
![image](https://github.com/monkeytypegame/monkeytype/assets/13181393/03fae076-d966-49e1-b1af-b20c7797de8b)
